src/lib/cardiac_mechanics/LinearSolverUtility.hpp

```diff
@@ -0,0 +1,320 @@
+/**
+ * @file LinearSolverUtility.hpp
+ * @brief Header file defining the LinearSolverUtility class for configuring
+ * linear solvers.
+ */
+
+#ifndef LINEAR_SOLVER_CONFIGURATION_HPP
+#define LINEAR_SOLVER_CONFIGURATION_HPP
+
+#include <deal.II/lac/solver_bicgstab.h>
+#include <deal.II/lac/solver_control.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/trilinos_precondition.h>
+
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+
+/**
+ * @class LinearSolverUtility
+ * @brief Utility class for configuring linear solvers.
+ * @tparam Scalar The scalar type used in the linear solver configuration
+ * (default is double).
+ */
+template <typename Scalar = double> class LinearSolverUtility {
```

Comment:
If I understand correctly, this class does not manage the solver itself. Instead, it provides methods to initialize a solver and a preconditioner and to use them, but their actual storing is demanded to the user of this class (i.e. the class that makes use of it).

Therefore, from a user's perspective, if I want to use this class, I would have to declare an instance of `LinearSolverUtility`, an instance of linear solver and an instance of preconditioner. Furthermore, the latter two would need to be compliant with the expectations of `LinearSolverUtility` (that is, they would need to be `unique_ptr`s). This is putting a lot of work (in relative terms) on the user.

A better alternative would be to directly declare the solver and preconditioner within `LinearSolverUtility`, and hide them from the user. The user would only have to specify the relevant settings (which type of solver and of preconditioner, tolerances and so on).

Furthermore, the `LinearSolverUtility` class might also take care of declaring and parsing the parameters associated to the linear solver and preconditioner, so that the user would not even have to worry about that.

As a general rule, wrappers such as this one should be designed in such a way that the user interface is simplified with respect to working without the wrapper. This simplification often comes in terms of reducing the amount of boilerplate code and of code in general needed to achieve a certain feature.

Reply:
I really appreciate this observation, thanks!